apis/compute/v1alpha2/types.go

```diff
@@ -69,14 +167,23 @@ type EKSClusterParameters struct {
 	// specify  up  to  5  security groups, but we recommend that you use a
 	// dedicated security group for your cluster control plane.
 
-	// VpcID of this EKS cluster.
-	VpcID string `json:"vpcId"`
+	// VPCID is the ID of the VPC.
+	VPCID string `json:"vpcId,omitempty"`
 
-	// SubnetIds of this EKS cluster.
-	SubnetIds []string `json:"subnetIds"`
+	// VPCIDRef references to a VPC to and retrieves its vpcId
+	VPCIDRef *VPCIDReferencerForEKSCluster `json:"vpcIdRef,omitempty" resource:"attributereferencer"`
```

Comment:
It looks like `VPCIDReferencerForEKSCluster` expects to operate on `EKSClusterClass` but this parameters struct is used in both `EKSClusterClass` and `EKSCluster`. This won't be an issue if we can avoid resolving references at the class level, but otherwise (if I follow correctly) I think we'd have to teach the resolver how to handle both the class and managed resource cases?

Reply:
We are avoiding resolving at class (claim) level.